researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -729,9 +738,20 @@ public void uploadTaskResult(@NonNull TaskResult taskResult) {
         uploadTaskResult(null, taskResult);
     }
 
+    /**
+     * @param context can be activity or application
+     * @param taskResult from the user doing the task
+     * @param associatedSchedule linked to the task result, for creating metadata and updating schedule on bridge
+     * @param shouldUpdateAssociatedScheduleOnBridge if true, schedule will be updated,
+     *                                               if false, it will only be used to create metadata
+     */
     @SuppressLint("RxLeakedSubscription")    // upload should run as long as it needs to, no early unsubscribe
-    @Override
-    public void uploadTaskResult(Context context, @NonNull TaskResult taskResult) {
+    public void uploadTaskResult(
```

Comment:
These parameters allow me to upload to s3 with proper metadata file without having to also update the activity on bridge (which we now do with Room and ScheduleRepository)

Reply:
Since Room and ScheduleRepository for SageResearch tasks and BridgeDataProvider for ResearchStack tasks, don't we always need to update the activity when calling this code path?

Reply:
Room and ScheduleRepository are decoupled from both RS tasks and SageResearch tasks.  They are just the source of truth for all ScheduleActivity objects in the app, and they always need to stay synced whether a RS task or a SR task was completed.  Therefore when either is complete, I just call ScheduleRepository.updateSchedule(schedule) and it will make the call to bridge and update the db.

Reply:
That is true and the right way to do. ScheduleRepo is in sageresearch-app-sdk and not android-sdk, so I think this only applies because we're using both SageResearch and ResearchStack. Is that right?

Reply:
Yeah, exactly.